GomokuClient/packages/gomoku-core/src/pages/HomeGame.tsx

```diff
@@ -47,6 +52,33 @@ export const HomeGame = () => {
       icon: <Users className="mr-3 h-5 w-5 text-[#bababa] sm:h-6 sm:w-6" />,
     }));
 
+  useEffect(() => {
+    if (isConnected && hubProxy) {
+      const unregister = registerEventHandlers({
+        onMatchingPlayerFound: async (gameId) => {
+          await router.navigate({
+            to: `/game/join/${gameId}`,
+          });
+        },
+        onOnlineUserCountChange: async (usersCount: number) => {
+          setOnlineUsersCount(usersCount);
+        },
+      });
+      return () => {
+        if (typeof unregister === "function") {
+          unregister();
+        }
+      };
+    }
+    return;
+  }, [isConnected, hubProxy, registerEventHandlers, router]);
+
+  useEffect(() => {
+    return () => {
+      hubProxy?.leaveQueue();
+    };
+  }, [hubProxy]);
+
```

Comment:
It looks better, but still this is something we should place inside tanstack-router ideally, if possible. We can use onLeave api. https://tanstack.com/router/latest/docs/framework/react/api/router/RouteOptionsType#onleave-property

Add the onLeave logic in the router configuration for the /home route:

```tsx
const router = createRouter({
  routes: [
    {
      path: '/home',
      element: <HomeGame />,
      onLeave: () => {
        // Ensure hubProxy leaves the queue when navigating away
        hubProxy?.leaveQueue();
      },
    },
    // other routes...
  ],
});
```

Reply:
Routing-related side effects, like leaving a queue when changing pages, are better managed in routing logic rather than the component lifecycle

Reply:
I can't use useSignalRConnection hook to acceess hubProxy and call required method. I added TODO to figure our later if we can do this